src/main/java/com/yelp/nrtsearch/server/luceneserver/facet/DrillSidewaysImpl.java

```diff
@@ -0,0 +1,363 @@
+    if (drillSidewaysDims != null) {
+      for (int i = 0; i < drillSidewaysDims.length; i++) {
+        dsDimMap.put(drillSidewaysDims[i], drillSideways[i]);
+      }
+    }
+
+    // Holds already computed Facets, since more
+    // than one dimension can share a single
+    // index field name.  We need one map for "normal" and
+    // another for SSDV facets because an app can index both
+    // into the same Lucene field (this is the default):
+    Map<String, Facets> indexFieldNameToFacets = new HashMap<String, Facets>();
+    Map<String, Facets> indexFieldNameToSSDVFacets = new HashMap<String, Facets>();
+
+    for (Facet facet : grpcFacets) {
+      String fieldName = facet.getDim();
+      FieldDef fd = dynamicFields.get(fieldName);
+      if (fd == null) {
+        throw new IllegalArgumentException(
+            String.format(
+                "field %s was not registered and was not specified as a dynamic field ",
+                fieldName));
+      }
+
+      FacetResult facetResult;
+      if (!(fd instanceof IndexableFieldDef)) { // TODO: Also enable facet support in Virtual Fields
+        throw new IllegalArgumentException(
+            String.format(
+                "field % not registered as an indexable field. Facets applicable only to indexable fields",
+                fieldName));
+      }
+      IndexableFieldDef indexableFieldDef = (IndexableFieldDef) fd;
+      if (!facet.getNumericRangeList().isEmpty()) {
+
+        if (indexableFieldDef.getFacetValueType()
+            != IndexableFieldDef.FacetValueType.NUMERIC_RANGE) {
+          throw new IllegalArgumentException(
+              String.format(
+                  "field %s was not registered with facet=numericRange",
+                  indexableFieldDef.getName()));
+        }
+        if (indexableFieldDef instanceof IntFieldDef || indexableFieldDef instanceof LongFieldDef) {
+          List<NumericRangeType> rangeList = facet.getNumericRangeList();
+          LongRange[] ranges = new LongRange[rangeList.size()];
+          for (int i = 0; i < ranges.length; i++) {
+            NumericRangeType numericRangeType = rangeList.get(i);
+            ranges[i] =
+                new LongRange(
+                    numericRangeType.getLabel(),
+                    numericRangeType.getMin(),
+                    numericRangeType.getMinInclusive(),
+                    numericRangeType.getMax(),
+                    numericRangeType.getMaxInclusive());
+          }
+
+          FacetsCollector c = dsDimMap.get(indexableFieldDef.getName());
+          if (c == null) {
+            c = drillDowns;
+          }
+
+          LongRangeFacetCounts longRangeFacetCounts =
+              new LongRangeFacetCounts(indexableFieldDef.getName(), c, ranges);
+          facetResult =
+              longRangeFacetCounts.getTopChildren(
+                  0,
+                  indexableFieldDef.getName(),
+                  facet.getPathsList().toArray(new String[facet.getPathsCount()]));
```

Comment:
Not needed in this review: Do we want to structurally put these in individual FieldDefs? `FacetResult getFacetResults(facet, dsDimMap, drillDowns`?